todomvc/assets/js/app.js

```diff
@@ -0,0 +1,440 @@
+
+  setFilter(filter) {
+    this.currentFilter = filter
+    
+    // Update filter links
+    this.filterLinks.forEach(link => {
+      link.classList.remove('selected')
+      if (link.getAttribute('href') === `#/${filter}` || (filter === 'all' && link.getAttribute('href') === '#/')) {
+        link.classList.add('selected')
+      }
+    })
+    
+    this.render()
+  }
+
+  getFilteredTodos() {
+    const todos = Object.values(this.todos)
+    
+    switch (this.currentFilter) {
+      case 'active':
+        return todos.filter(todo => !todo.completed)
+      case 'completed':
+        return todos.filter(todo => todo.completed)
+      default:
+        return todos
+    }
+  }
+
+  render() {
+    const todos = Object.values(this.todos)
+    const filteredTodos = this.getFilteredTodos()
+    const activeTodos = todos.filter(todo => !todo.completed)
+    const completedTodos = todos.filter(todo => todo.completed)
+    
+    // Show/hide main and footer sections
+    const hasTodos = todos.length > 0
+    this.mainSection.style.display = hasTodos ? 'block' : 'none'
+    this.footerSection.style.display = hasTodos ? 'block' : 'none'
+    
+    // Update toggle all checkbox
+    this.toggleAllCheckbox.checked = activeTodos.length === 0 && todos.length > 0
+    
+    // Render todo list
+    this.todoList.innerHTML = filteredTodos.map(todo => this.renderTodoItem(todo)).join('')
+    
+    // Update todo count
+    const activeCount = activeTodos.length
+    const itemText = activeCount === 1 ? 'item' : 'items'
+    this.todoCount.innerHTML = `<strong>${activeCount}</strong> ${itemText} left`
+    
+    // Show/hide clear completed button
+    this.clearCompletedBtn.style.display = completedTodos.length > 0 ? 'block' : 'none'
+    
+    // Bind todo item events
+    this.bindTodoEvents()
+  }
+
+  renderTodoItem(todo) {
+    return `
+      <li data-id="${todo.id}" class="${todo.completed ? 'completed' : ''}">
+        <div class="view">
+          <input class="toggle" type="checkbox" ${todo.completed ? 'checked' : ''}>
+          <label>${this.escapeHtml(todo.text)}</label>
+          <button class="destroy"></button>
+        </div>
+        <input class="edit" value="${this.escapeHtml(todo.text)}">
+      </li>
+    `
+  }
+
+  bindTodoEvents() {
+    // Toggle todo
+    this.todoList.querySelectorAll('.toggle').forEach(checkbox => {
+      checkbox.addEventListener('change', (e) => {
+        const li = e.target.closest('li')
+        const id = li.dataset.id
+        this.updateTodo(id, { completed: e.target.checked })
+      })
+    })
+    
+    // Delete todo
+    this.todoList.querySelectorAll('.destroy').forEach(button => {
+      button.addEventListener('click', (e) => {
+        const li = e.target.closest('li')
+        const id = li.dataset.id
+        this.deleteTodo(id)
+      })
+    })
+    
+    // Edit todo
+    this.todoList.querySelectorAll('label').forEach(label => {
+      label.addEventListener('dblclick', (e) => {
```

Comment:
I realise this comes from the TODO MVC examples, but using `dblclick` exclusively makes it mostly unusable in mobile and for keyboard users.